caddyhttp/fastcgi/fastcgi.go

```diff
@@ -73,9 +82,33 @@ func (h Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) (int, error)
 
 			// Connect to FastCGI gateway
 			network, address := rule.parseAddress()
-			fcgiBackend, err := Dial(network, address)
-			if err != nil {
-				return http.StatusBadGateway, err
+
+			var fcgiBackend *FCGIClient
+			var mut *sync.Mutex
+			ok := false
+			if usePersistentFcgiConnections {
+				// re use connection, if possible
+				if persistentConnections == nil {
+					persistentConnections = make(map[string]*FCGIClient)
+					poorMansSerialisation = make(map[string]*sync.Mutex)
+				}
+				mut, ok = poorMansSerialisation[network+address]
+				if !ok || mut == nil {
```

Comment:
If you change the code below to use `delete()` then you can reduce this just `if !ok`